src/hooks/useUndoRedoState.ts

```diff
@@ -19,26 +21,27 @@ export default function useUndoRedoState<T>(initialValue?: T): [T | undefined, S
   const [stack, setStack] = useState<(T | undefined)[]>([initialValue]);
   const [stackIndex, setStackIndex] = useState<number>(0);
 
-  const data = useMemo<T | undefined>(() => stack[stackIndex], [stackIndex, stack]);
+  const data = useMemo<T | undefined>(() => _.cloneDeep(stack[stackIndex]), [stackIndex, stack]);
 
   const setData = useCallback(
-    (value: T) => {
+    (input: Func<T> | T) => {
+      const value = typeof input === 'function' ? (input as Func<T>)(_.cloneDeep(stack[stackIndex])) : input;
```

Comment:
Can this be `(input as Func<T>)(data)` since it is memoized above?

Reply:
Yes it can, I had it that way then decided against it.. will use that. Thanks for all the reviews!